eclipse-scout-core/src/table/Table.ts

```diff
@@ -6156,8 +6466,90 @@ export class Table extends Widget implements TableModel, Filterable<TableRow> {
   }
 
   isCustomizable(): boolean {
-    // TODO bsh [js-table] Delegate to this.customizer
-    return false;
+    return !!this.customizer;
+  }
+
+  setUiPreferencesEnabled(uiPreferencesEnabled: boolean) {
+    this.setProperty('uiPreferencesEnabled', !!uiPreferencesEnabled);
+  }
+
+  _setUiPreferencesEnabled(uiPreferencesEnabled: boolean) {
+    this._setProperty('uiPreferencesEnabled', uiPreferencesEnabled);
+    if (uiPreferencesEnabled) {
+      // Save current state as "factory defaults". User filters are not included.
+      this.setInitialUiPreferences(uiPreferences.createTablePreferenceProfile(this));
+
+      // If there is a stored GLOBAL profile, apply it now
+      let prefs = uiPreferences.getTablePreferences(this);
+      uiPreferences.applyTablePreferences(this, prefs, UiPreferences.TABLE_PREFERENCE_PROFILE_ID_GLOBAL);
+
+      // Install a table listener that stores all changes into the GLOBAL profile.
+      // This is only done _after_ applying the initial state, so that no events were triggered.
+      uiPreferences.installTableListener(this);
+    } else {
+      // Uninstall listener
+      uiPreferences.uninstallTableListener(this);
+    }
+  }
+
+  setInitialUiPreferences(initialUiPreferences: TableClientUiPreferenceProfileDo) {
+    this._initialUiPreferences = initialUiPreferences;
+  }
+
+  get initialUiPreferences(): TableClientUiPreferenceProfileDo {
+    return this._initialUiPreferences;
+  }
+
+  /**
+   * Applies the {@link initialUiPreferences} and clears the stored {@link UiPreferences#TABLE_PREFERENCE_PROFILE_ID_GLOBAL} profile.
+   * If the initial ui preferences were not previously set, this method has no effect.
+   */
+  resetToInitialUiPreferences() {
+    if (!this._initialUiPreferences) {
+      return;
+    }
+    uiPreferences.applyTablePreferencesProfile(this, this._initialUiPreferences, {
+      applyUserFilters: true
+    });
+    uiPreferences.clearGlobalTablePreferenceProfile(this);
+  }
+
+  applyColumnPreferences(columnPreferences: TableColumnClientUiPreferenceDo[]) {
+    let columnPreferencesMap = new Map(arrays.ensure(columnPreferences).map(pref => [pref.columnId, pref]));
+    let newColumns = this.columns
+      .filter(c => !c.guiOnly)
```

Comment:
maybe map to \[column, prefs] first to simplify the sort and map functions? 
.map(column => \[column, columnPreferencesMap.get(column.buildUuid())] as \[Column<any>, TableColumnClientUiPreferenceDo])